backend/module/eCampCore/src/Service/InvitationService.php

```diff
@@ -19,21 +20,46 @@ class InvitationService {
 
     public function __construct(ServiceUtils $serviceUtils, AuthenticationService $authenticationService, UserService $userService) {
         $this->authenticationService = $authenticationService;
-        $this->campCollaborationRepository = $serviceUtils->emGetRepository(CampCollaboration::class);
+        /** @var CampCollaborationRepository $entityRepository */
+        $entityRepository = $serviceUtils->emGetRepository(CampCollaboration::class);
+        $this->campCollaborationRepository = $entityRepository;
         $this->userService = $userService;
     }
 
-    public function findInvitation(string $inviteKey): ?CampCollaboration {
-        return $this->campCollaborationRepository->findByInviteKey($inviteKey);
+    /**
+     * @throws EntityNotFoundException
+     * @throws NoAccessException
+     * @throws NonUniqueResultException
+     */
+    public function findInvitation(string $inviteKey): ?Invitation {
+        $campCollaboration = $this->campCollaborationRepository->findByInviteKey($inviteKey);
+        if (null == $campCollaboration) {
+            return null;
+        }
+        $camp = $campCollaboration->getCamp();
+        $userDisplayName = null;
+        $userAlreadyInCamp = null;
+        $userId = $this->authenticationService->getIdentity();
+        if (null != $userId) {
+            /** @var User $user */
+            $user = $this->userService->fetch($userId);
+            $userDisplayName = $user->getDisplayName();
+            $existingCampCollaboration = $this->campCollaborationRepository->findByUserAndCamp($user, $camp);
+            $userAlreadyInCamp = null != $existingCampCollaboration;
```

Comment:
```suggestion
            $userAlreadyInCamp = null != $existingCampCollaboration && $existingCampCollaboration->isEstablished();
```

Beim zweiten Einladen (CampCollaboration.Status == Left) ist User bereits gesetzt.
Dann wird CampCollaboration gefunden, hat aber status "invited".
$userAlreadyInCamp meint wohl nur Established CampCollaborations.

![image](https://user-images.githubusercontent.com/470237/108624460-615d4500-7445-11eb-9f4e-51a691b151d6.png)

Obwohl CampCollaboration.Status == "invited"